src/main/java/io/jenkins/plugins/pipeline_elasticsearch_logs/write/fluentd/FluentdWriter.java

```diff
@@ -0,0 +1,268 @@
+
+    private static final Logger LOGGER = Logger.getLogger(FluentdWriter.class.getName());
+
+    private transient FluentLogger fluentd;
+    private transient FluentdErrorHandler errorHandler;
+    private transient boolean configChanged;
+
+    private final static int SYS_LOG_FREQUENCY = 2000; //TODO: 10*1000;
+
+    private final static int DEFAULT_TIMEOUT_MILLIS = 3000;
+    private final static int DEFAULT_RETRY_MILLIS = 1000;
+    private final static int DEFAULT_BUFFER_CAPACITY = 1048576;
+
+    private String tagPrefix;
+    private String tag;
+
+    private String host;
+    private int port;
+    private int retryMillis = DEFAULT_RETRY_MILLIS;
+    private int timeoutMillis = DEFAULT_TIMEOUT_MILLIS;
+    private int bufferCapacity = DEFAULT_BUFFER_CAPACITY;
+
+    @DataBoundConstructor
+    public FluentdWriter() throws URISyntaxException {
+    }
+
+    public String getTagPrefix() {
+        return tagPrefix;
+    }
+
+    @DataBoundSetter
+    public void setTagPrefix(String tagPrefix) {
+        this.tagPrefix = tagPrefix;
+        this.configChanged = true;
+    }
+
+    public String getTag() {
+        return tag;
+    }
+
+    @DataBoundSetter
+    public void setTag(String tag) {
+        this.tag = tag;
+        this.configChanged = true;
+    }
+
+    public String getHost() {
+        return host;
+    }
+
+    @DataBoundSetter
+    public void setHost(String host) {
+        this.host = host;
+        this.configChanged = true;
+    }
+
+    public int getPort() {
+        return port;
+    }
+
+    @DataBoundSetter
+    public void setPort(int port) {
+        if(port < 0 || port > 65535) throw new IllegalArgumentException("Port out of range: " + port);
+        this.port = port;
+        this.configChanged = true;
+    }
+
+    public int getRetryMillis() {
+        return retryMillis;
+    }
+
+    @DataBoundSetter
+    public void setRetryMillis(int retryMillis) {
+        if (retryMillis < 0)
+            retryMillis = 0;
+        this.retryMillis = retryMillis;
+        this.configChanged = true;
+    }
+
+    public int getTimeoutMillis() {
+        return timeoutMillis;
+    }
+
+    @DataBoundSetter
+    public void setTimeoutMillis(int timeoutMillis) {
+        if(timeoutMillis < 0) throw new IllegalArgumentException("timeoutMillis less than 0");
+        this.timeoutMillis = timeoutMillis;
+        this.configChanged = true;
+    }
+
+    public int getBufferCapacity() {
+        return bufferCapacity;
+    }
+
+    @DataBoundSetter
+    public void setBufferCapacity(int bufferCapacity) {
+        if (bufferCapacity < 0)
+            bufferCapacity = DEFAULT_BUFFER_CAPACITY;
+        this.bufferCapacity = bufferCapacity;
+        this.configChanged = true;
+    }
+
+
+    @Extension
+    @Symbol("fluentd")
+    public static class DescriptorImpl extends ElasticSearchWriteAccessDescriptor {
+        @Override
+        public String getDisplayName() {
+            return "Fluentd Elasticsearch Writer";
```

Comment:
Can't fluentd also forward to other targets than ES?
```suggestion
            return "Fluentd Writer";
```

Reply:
there can be a fluentd or a fluentbit and they can forward to many other systems or even file system